### ISSUE TITLE
Support for Nvidia Tesla T4 

### ISSUE THREAD

**Issue by gautamkmr** [pytorch/pytorch#16733]
## 🚀 Feature
Current CMakeList.txt has support up to Volta SM=70. It would be great if we have support for T4 as well. I believe this architecture is more for inference than training but still a good use case to consider. 
 I believe it has to accommodate Tesla T4 by supporting cuda compute ability 7.5   https://github.com/pytorch/pytorch/blob/9477a5d9c80d96e588e2e0e80320efcccbe83414/aten/src/ATen/CMakeLists.txt#L267

## Motivation
Support for new GPU  T4.
https://www.nvidia.com/en-us/data-center/tesla-t4/

## Pitch

https://www.nvidia.com/en-us/data-center/tesla-t4/
## Alternatives

N/A
## Additional context
https://www.nvidia.com/en-us/data-center/tesla-t4/
https://en.wikipedia.org/wiki/Nvidia_Tesla .

**Comment by soumith** [comment]
we actually support sm_75 in our CUDA10 binaries. The pointed code is for a fakelink cufft library, that shouldn't really matter for any real codepath.

**Comment by soumith** [comment]
the relevant flags are at: https://github.com/pytorch/pytorch/blob/master/cmake/Modules_CUDA_fix/upstream/FindCUDA/select_compute_arch.cmake

**Comment by soumith** [comment]
I'm totally wrong. looks like select_compute_arch.cmake doesn't add support for sm_75, we need to update it.

However, considering sm_70 is binary-compatible with sm_75, the code should run fine (if you are running CUDA10 pytorch binaries). Is this not the case?

**Comment by gautamkmr** [comment]
"However, considering sm_70 is binary-compatible with sm_75, the code should run fine (if you are running CUDA10 pytorch binaries). Is this not the case?" 

Yeah I can give a sort on testing, but in past we had experience where pytorch compiled with sm_30 was not working on sm_35.

**Comment by soumith** [comment]
sm_30 and sm_35 are special. sm_30 is binary-compatible with sm_35, but sm_35 introduced instructions that made it significantly faster for some kernels (and also not compatible backwards with sm_30).

**Comment by soumith** [comment]
this thread suggests that the sm_70 / cuda10 version of PyTorch should work with T4: https://github.com/pytorch/pytorch/issues/14260#issuecomment-451987294

let me know how it goes. Either ways, we will introduce sm_75 to the relevant files.

**Comment by bhavinthaker** [comment]
What is the expected date/release by which sm_75 support will be added?

I think adding the relevant flags helps ensure that not only the functionality works but additionally it also helps ensure that the performance is the best possible. Please correct me if required.

**Comment by soumith** [comment]
sm_75 is shipped with PyTorch's CUDA10 binaries. It's reflected in this line: https://github.com/pytorch/builder/blob/master/conda/pytorch-1.0.1/build.sh#L32

PyTorch 1.0.1 should be good to go, as long as you select and install the CUDA10 version of v1.0.1 via the selector on pytorch.org

**Comment by soumith** [comment]
@bhavinthaker this should be part of 1.1 (sorry for my earlier message which was incorrect, I deleted it)

**Comment by bhavinthaker** [comment]
@soumith Any tentative date for PyTorch 1.1 release? This will help us plan the AWS Deep Learning AMI release accordingly.

**Comment by soumith** [comment]
@bhavinthaker target date April 30th, for F8

**Comment by soumith** [comment]
this is now live, and ships sass code for 7.5 (if you use the CUDA10 builds)

**Comment by soumith** [comment]
the commit that fixed it was https://github.com/pytorch/pytorch/pull/19392

**Comment by lingdoc** [comment]
After installing Pytorch with the following instructions (as per https://pytorch.org/):
```
pip install torch==1.6.0+cu101 torchvision==0.7.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html
```

I'm getting this error message:
```
GeForce RTX 2080 Ti with CUDA capability sm_75 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_70.
```
Is sm_75 code not included for cuda 10.1? Training still runs, but it seems not to take advantage of the instruction set.
